public/app/features/plugins/admin/components/UpdateAllModal.tsx

```diff
@@ -0,0 +1,283 @@
+    marginTop: theme.spacing(3),
+  }),
+  noPluginsMessage: css({
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    height: '100%',
+  }),
+  tableContainer: css({
+    overflowY: 'auto',
+    overflowX: 'hidden',
+    height: theme.spacing(32),
+  }),
+  modalContainer: css({
+    height: theme.spacing(41),
+  }),
+});
+
+type ModalBodyProps = {
+  plugins: CatalogPlugin[];
+  inProgress: boolean;
+  selectedPlugins?: Set<string>;
+  onCheckboxChange: (id: string) => void;
+  errorMap: Map<string, UpdateError>;
+};
+
+const ModalBody = ({ plugins, inProgress, selectedPlugins, onCheckboxChange, errorMap }: ModalBodyProps) => {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.modalContainer}>
+      {plugins.length === 0 ? (
+        <EmptyState
+          variant="completed"
+          message={t('plugins.catalog.update-all.all-plugins-updated', 'All plugins updated!')}
+        />
+      ) : (
+        <>
+          <div>
+            <Trans i18nKey="plugins.catalog.update-all.header">The following plugins have update available</Trans>
+          </div>
+          <div className={styles.tableContainer}>
+            <table className={styles.table}>
+              <thead className={styles.header}>
+                <tr>
+                  <th>
+                    <Trans i18nKey="plugins.catalog.update-all.update-header">Update</Trans>
+                  </th>
+                  <th>
+                    <Trans i18nKey="plugins.catalog.update-all.name-header">Name</Trans>
+                  </th>
+                  <th>
+                    <Trans i18nKey="plugins.catalog.update-all.installed-header">Installed</Trans>
+                  </th>
+                  <th>
+                    <Trans i18nKey="plugins.catalog.update-all.available-header">Available</Trans>
+                  </th>
+                  <th></th>
+                </tr>
+              </thead>
+              <tbody>
+                {plugins.map(({ id, name, installedVersion, latestVersion }: CatalogPlugin) => (
+                  <tr key={id} className={styles.tableRow}>
+                    <td>
+                      <Checkbox onChange={() => onCheckboxChange(id)} value={selectedPlugins?.has(id)} />
+                    </td>
+                    <td>{name}</td>
+                    <td>{installedVersion}</td>
+                    <td>{latestVersion}</td>
+                    <td className={styles.icon}>{getIcon({ id, inProgress, errorMap, selectedPlugins })}</td>
+                  </tr>
+                ))}
+              </tbody>
+            </table>
+          </div>
+          {config.pluginAdminExternalManageEnabled && config.featureToggles.managedPluginsInstall && (
+            <footer className={styles.footer}>
+              <Trans i18nKey="plugins.catalog.update-all.cloud-update-message">
+                * It may take a few minutes for the plugins to be available for usage.
```

Comment:
> It may take a few minutes for the plugins to be available for usage

Given the plugins themselves will remain available, maybe we should change this to:
- It may take a few minutes for the new versions to be available for usage.